### DISCUSSION TITLE
<details> block

### DISCUSSION THREAD

**Discussion by e2045** [simonhaenisch/md-to-pdf#117] [Q&A] [answered]
Hi,

i must write some md files with details block. And the attribute "open" is not required. Example:

```
Some text

<details>
    <summary>Click here for more details </summary>

Info 1

Info 2

Info 3

</details>
```

I knew without "open" the result after conversion is bad. All contents of details block disappeared. 

![grafik](https://user-images.githubusercontent.com/51294439/155314542-075887a0-236a-4839-bf11-01a69f2aa8bb.png)

Do you have any idea how to show contents of details block in pdf?

Thank you in advance. 

My code:

```
function converting(file){
    //converting md to pdf
    (async () => {
        let pdf = await mdToPdf(
            {path: file},
            {pdf_options: {
                printBackground: true,
                format: 'a4',
                margin: {
                    top: '30mm',
                    right: '20mm',
                    bottom: '30mm',
                    left: '20mm',
                },
            }
        }).catch(console.error);

        if (pdf) {
            let newFile = file.substring(0, (file.length - 3)) + ".pdf"
            fs.writeFileSync(newFile, pdf.content);
            console.log(newFile + ' is written');
        }

    })();
}
```

My file:
[detailsblock.md](https://github.com/simonhaenisch/md-to-pdf/files/8124086/detailsblock.md)
[detailsblock.pdf](https://github.com/simonhaenisch/md-to-pdf/files/8124126/detailsblock.pdf)

**Comment by simonhaenisch** [comment, marked as answer]
Hey, you can also inject a script into the page that does this for you

```markdown
---
script:
  content: "document.querySelectorAll('details').forEach(details => { details.open = true })"
---

<details>I'm in the details</details>
```

I just realized that the `script` option is not mentioned in the docs yet. See here: https://github.com/simonhaenisch/md-to-pdf/blob/3e45dc7634aa0cc5a7fb128b1836aea9613e6b6d/src/lib/config.ts#L84-L89

**Reply by e2045** [reply]
wow thank you. It works like a charm 😄
